network-config-analyzer/IstioTrafficResourcesYamlParser.py

```diff
@@ -0,0 +1,436 @@
+                self.warning(f'Missing port for service {dest} in the VirtualService {vs.full_name()}')
+            parsed_route.add_destination(service, target_port)
+
+            if item.get('headers'):
+                self.warning(f'"headers" are not yet supported in HTTPRouteDestination '
+                             f'(referenced in the VirtualService {vs.full_name()}). Ignoring them.')
+
+    def parse_service(self, dest, vs):
+        """
+        Parse Destination resource of the VirtualService and return a service corresponding to it
+        :param dict dest: the destination resource, as defined in
+        https://istio.io/latest/docs/reference/config/networking/virtual-service/#Destination
+        :param vs: the VirtualService contining this destinatnion
+        :return K8sService: the service corresponding to the given destination resource
+        """
+        host = dest.get('host')
+        if not host:
+            return None
+        # according to https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+        # check for <service-name>.<namespace-name>.svc.cluster.local
+        splitted_host = host.split('.')
+        if len(splitted_host) == 5 and splitted_host[2] == 'svc' and splitted_host[3] == 'cluster' \
+                and splitted_host[4] == 'local':
+            service_name = splitted_host[0]
+            namespace = self.peer_container.get_namespace(splitted_host[1])
+        else:
+            service_name = host
+            namespace = vs.namespace
+        return self.peer_container.get_service_by_name_and_ns(service_name, namespace)
+
+    def make_allowed_connections(self, vs, host_dfa):
+        """
+        Create allowed connections of the given VirtualService
+        :param VirtualService vs: the given VirtualService and the given hosts
+        :param MinDFA host_dfa: the hosts attribute
+        :return: TcpLikeProperties with TCP allowed connections
+        """
+        allowed_conns = None
+        for http_route in vs.http_routes:
+            for dest in http_route.destinations:
+                conns = self._make_tcp_like_properties(dest.port, dest.service.target_pods, http_route.uri_dfa,
+                                                       host_dfa, http_route.method_dfa)
+                if not allowed_conns:
+                    allowed_conns = conns
+                else:
+                    allowed_conns |= conns
+        return allowed_conns
+
+    def create_istio_traffic_policies(self):
+        """
+        Create IngressPolicies according to the parsed Gateways and VirtualServices
+        :return list[IngressPolicy]: the resulting policies
+        """
+
+        if not self.gateways:
+            self.warning('No valid Gateways found. Ignoring istio ingress traffic')
+            return []
+        if not self.virtual_services:
+            self.warning('No valid VirtualServices found. Ignoring istio ingress traffic')
+            return []
```

Comment:
We probably need a stronger check: is there a Gateway with no VirtualServices attached?